library/src/main/kotlin/com/chuckerteam/chucker/internal/ui/MainActivity.kt

```diff
@@ -148,15 +168,16 @@ internal class MainActivity :
                 showDialog(
                     getClearDialogData(),
                     onPositiveClick = {
-                        viewModel.clearTransactions()
+                        if (selectedTransactions.isNotEmpty()) viewModel.clearSelectedTransactions(selectedTransactions) else viewModel.clearTransactions()
```

Comment:
Can we move this logic inside hte ViewModel? Let's just have a `viewModel.clearTransactions(selectedTransactions)` that behaves differently wether the `selectedTransactions` is empty or not